src/legacy/core_plugins/kibana/public/discover/embeddable/search_embeddable.ts

```diff
@@ -178,52 +185,51 @@ export class SearchEmbeddable extends Embeddable<SearchInput, SearchOutput>
     if (this.subscription) {
       this.subscription.unsubscribe();
     }
+    timefilter.off('autoRefreshFetch', this.fetch);
   }
 
   private initializeSearchScope() {
-    const searchScope: SearchScope = this.$rootScope.$new();
+    const searchScope: SearchScope = (this.searchScope = this.$rootScope.$new());
 
     searchScope.description = this.savedSearch.description;
-    searchScope.searchSource = this.savedSearch.searchSource;
     searchScope.inspectorAdapters = this.inspectorAdaptors;
 
-    const timeRangeSearchSource = searchScope.searchSource.create();
+    const { searchSource } = this.savedSearch;
+    const indexPattern = (searchScope.indexPattern = searchSource.getField('index'));
+
+    const timeRangeSearchSource = searchSource.create();
     timeRangeSearchSource.setField('filter', () => {
       if (!this.searchScope || !this.input.timeRange) {
         return;
       }
-      return getTime(this.searchScope.searchSource.getField('index'), this.input.timeRange);
+      return getTime(indexPattern, this.input.timeRange);
     });
 
-    this.filtersSearchSource = searchScope.searchSource.create();
+    this.filtersSearchSource = searchSource.create();
     this.filtersSearchSource.setParent(timeRangeSearchSource);
 
-    searchScope.searchSource.setParent(this.filtersSearchSource);
+    searchSource.setParent(this.filtersSearchSource);
 
     this.pushContainerStateParamsToScope(searchScope);
 
-    searchScope.setSortOrder = sortPair => {
-      searchScope.sort = sortPair;
-      this.updateInput({ sort: searchScope.sort });
+    searchScope.setSortOrder = sort => {
+      this.updateInput({ sort });
     };
 
     searchScope.addColumn = (columnName: string) => {
       if (!searchScope.columns) {
         return;
       }
-      this.savedSearch.searchSource.getField('index').popularizeField(columnName, 1);
+      indexPattern.popularizeField(columnName, 1);
       columnActions.addColumn(searchScope.columns, columnName);
-      searchScope.columns = searchScope.columns;
       this.updateInput({ columns: searchScope.columns });
     };
 
     searchScope.removeColumn = (columnName: string) => {
       if (!searchScope.columns) {
         return;
       }
-      this.savedSearch.searchSource.getField('index').popularizeField(columnName, 1);
```

Comment:
Why were we "popularizing" a field when it was removed? Doesn't make sense to me.